tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,773 @@
+    );
+});
+
+test( 'Navigating Backward by Month', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'September ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '.sl-calendar .prev' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'August ' + currentYear,
+        'The previous month is set correctly'
+    );
+});
+
+test( 'Navigating Forward by Year', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+            viewMode="months"
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+
+    this.$( '.sl-calendar .next' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        currentYear + 1,
+        'The next year is set correctly'
+    );
+});
+
+test( 'Navigating Backward by Year', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+            viewMode="months"
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+
+    this.$( '.sl-calendar .prev' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        currentYear - 1,
+        'The previous year is set correctly'
+    );
+});
+
+test( 'Navigating Forward by Decade', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
```

Comment:
What role does this serve in this test? The component will still render as expected, especially within the context of this test, without this.